integr8ly/cloud-resource-operator
# Add aws postgres metric provider

## Overview

Jira: https://issues.redhat.com/browse/INTLY-7224

## Verification
**Note :** This work contains redis metric provider, which is just boilerplate and will be handled in https://issues.redhat.com/browse/INTLY-7740

- run `make cluster/prepare`
- run `make cluster/seed/managed/postgres`
- run `make run`
- wait a few minutes for the instance to begin provisioning before cloud watch metrics are exposed
- verify the following debug statements are logged : 
```
DEBU[0008] {free_storage_space_average map[clusterID:hcd-rhmi-clouds-6pdvc instanceID:hcdrhmiclouds6pdvccloudresourceoper-eu6t namespace:cloud-resource-operator productName:productName resourceID:example-postgres strategy:aws-rds] 2.0676653056e+10}  controller=controller_cloudmetrics
DEBU[0008] {free_storage_space_maximum map[clusterID:hcd-rhmi-clouds-6pdvc instanceID:hcdrhmiclouds6pdvccloudresourceoper-eu6t namespace:cloud-resource-operator productName:productName resourceID:example-postgres strategy:aws-rds] 2.0710719488e+10}  controller=controller_cloudmetrics
DEBU[0008] {free_storage_space_minimum map[clusterID:hcd-rhmi-clouds-6pdvc instanceID:hcdrhmiclouds6pdvccloudresourceoper-eu6t namespace:cloud-resource-operator productName:productName resourceID:example-postgres strategy:aws-rds] 2.0659613696e+10}  controller=controller_cloudmetrics

```
